### PR TITLE
add rust major mode

### PR DESCRIPTION
**Pull request by MarcoIeni**
Add rust for issue #98
The [rust analyzer](https://marketplace.visualstudio.com/items?itemName=matklad.rust-analyzer) extension is required.
I haven't added some of the major mode key bindings described in [conventions](https://vspacecode.github.io/docs/conventions) because they didn't work for me.

I might add more key bindings in the future, but it's enough to start in my opinion.

### CODE REVIEW

**Comment by stevenguh** [comment]
I am wondering is there any reason the official LSP is not used as well? It seems like the official rust extension has an option to use rust-analyzer as LSP

File: package.json [hunk 1]
```diff
@@ -3740,4 +3809,4 @@
 		"jsonc-parser": "^2.3.0",
 		"lodash": "^4.17.19"
 	}
-}
+}
```
Comment (stevenguh) [inline package.json:3812]: Undo removal of newline

Comment (MarcoIeni) [reply package.json]: done, sorry 😅

**Comment by MarcoIeni** [comment]
Yes, the rust analyzer extension is what everyone uses.

File: package.json [hunk 2]
```diff
@@ -3577,6 +3577,75 @@
...
+														"name": "Rust analyzer: reload workspace",
+														"type": "command",
+														"command": "rust-analyzer.reloadWorkspace"
+													},
+													{
+														"key": "v",
+														"name": "Rust analyzer: Show version",
+														"type": "command",
+														"command": "rust-analyzer.serverVersion"
+													}
```
Comment (stevenguh) [inline package.json:3638]: I think the ordering is lower case first then upper case

Comment (MarcoIeni) [reply package.json]: fixed in 47b5f19

**Pull request merged**